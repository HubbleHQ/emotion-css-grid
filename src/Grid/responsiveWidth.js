/** @jsx jsx */
import { css } from '@emotion/core';

import config from './GridConfig';

const { mobileBreakpoint } = config;

/* @param {number} column width expressed as a percentage
 * The responsiveWidth CSS describes how columns should
 * always collapse to a single-column layout (100% width)
 * on mobile sizes. At any size above a mobile range,
 * the column's width will be set by whatever numeric
 * percentage value argument that is passed in.
 */
const responsiveWidth = widthValue => css`
  width: 100%;
  @media (min-width: ${mobileBreakpoint}px) {
    width: ${widthValue}%;
  }
`;

export default responsiveWidth;
