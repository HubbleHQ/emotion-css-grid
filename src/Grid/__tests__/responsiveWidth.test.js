import responsiveWidth from '../responsiveWidth';

describe('Grid : responsiveWidth(width, mobileBreakpoint) method', () => {
  it('should render CSS as expected when passed width and breakpoint values', () => {
    const test = responsiveWidth(50, 640);
    expect(test).toMatchSnapshot();
  });
  it('should return an Emotion object', () => {
    const test = responsiveWidth(33.333, 800);
    expect(typeof test).toBe('object');
  });
});
