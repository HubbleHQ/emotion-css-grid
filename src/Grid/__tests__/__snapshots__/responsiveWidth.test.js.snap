// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid : responsiveWidth(width, mobileBreakpoint) method should render CSS as expected when passed width and breakpoint values 1`] = `
Object {
  "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInJlc3BvbnNpdmVXaWR0aC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFXNkQiLCJmaWxlIjoicmVzcG9uc2l2ZVdpZHRoLmpzIiwic291cmNlc0NvbnRlbnQiOlsiLyoqIEBqc3gganN4ICovXG5pbXBvcnQgeyBjc3MgfSBmcm9tICdAZW1vdGlvbi9jb3JlJztcblxuLyogQHBhcmFtIHtudW1iZXJ9IGNvbHVtbiB3aWR0aCBleHByZXNzZWQgYXMgYSBwZXJjZW50YWdlXG4gKiBAcGFyYW0ge251bWJlcn0gbW9iaWxlIGJyZWFrcG9pbnQsIGV4cHJlc3NlZCBpbiBwaXhlbHNcbiAqIFRoZSByZXNwb25zaXZlV2lkdGggQ1NTIGRlc2NyaWJlcyBob3cgY29sdW1ucyBzaG91bGRcbiAqIGFsd2F5cyBjb2xsYXBzZSB0byBhIHNpbmdsZS1jb2x1bW4gbGF5b3V0ICgxMDAlIHdpZHRoKVxuICogb24gbW9iaWxlIHNpemVzLiBBdCBhbnkgc2l6ZSBhYm92ZSBhIG1vYmlsZSByYW5nZSxcbiAqIHRoZSBjb2x1bW4ncyB3aWR0aCB3aWxsIGJlIHNldCBieSB3aGF0ZXZlciBudW1lcmljXG4gKiBwZXJjZW50YWdlIHZhbHVlIGFyZ3VtZW50IHRoYXQgaXMgcGFzc2VkIGluLlxuICovXG5jb25zdCByZXNwb25zaXZlV2lkdGggPSAod2lkdGhWYWx1ZSwgbW9iaWxlQnJlYWtwb2ludCkgPT4gY3NzYFxuICB3aWR0aDogMTAwJTtcbiAgQG1lZGlhIChtaW4td2lkdGg6ICR7bW9iaWxlQnJlYWtwb2ludH1weCkge1xuICAgIHdpZHRoOiAke3dpZHRoVmFsdWV9JTtcbiAgfVxuYDtcblxuZXhwb3J0IGRlZmF1bHQgcmVzcG9uc2l2ZVdpZHRoO1xuIl19 */",
  "name": "1yytwl3",
  "next": undefined,
  "styles": "width:100%;@media (min-width:640px){width:50%;}",
}
`;
